spec/controllers/v2/tailorings_controller_spec.rb

```diff
@@ -199,24 +197,83 @@
         end
 
         it 'results in 404 error' do
-          get :tailoring_file, params: extra_params.merge(parents: [:policy], format: :xml)
+          get :tailoring_file, params: extra_params.merge(parents: [:policy], format: format)
 
           expect(response).to have_http_status :not_found
         end
       end
     end
 
-    context 'authorized via cert_auth' do
-      before do
-        allow(controller).to receive(:rbac_allowed?).and_call_original
-        allow(controller).to receive(:any_inventory_hosts?).and_return(true)
+    context 'XCCDF' do
+      let(:format) { :xml }
+      let(:tailoring_file) { Nokogiri::XML(response.body).remove_namespaces! }
+
+      let(:groups) do
+        tailoring_file.xpath(
+          '//Profile/select[starts-with(@idref, "xccdf_org.ssgproject.content_rule_group")]/@idref'
+        ).map(&:value)
+      end
+
+      let(:selected_rules) do
+        tailoring_file.xpath(
+          '//Profile/select[
+              starts-with(
+                @idref, "xccdf_org.ssgproject.content_rule_"
+              ) and not(contains(@idref, "rule_group"))
+           ][@selected="true"]/@idref'
+        ).map(&:value)
+      end
+
+      let(:deselected_rules) do
+        tailoring_file.xpath(
+          '//Profile/select[
+              starts-with(
+                @idref, "xccdf_org.ssgproject.content_rule_"
+              ) and not(contains(@idref, "rule_group"))
+           ][@selected="false"]/@idref'
+        ).map(&:value)
+      end
+
+      let(:values) do
+        tailoring_file.xpath('//Profile/set-value').each_with_object({}) do |value, value_override|
```

Comment:
The internal variable should be generic, the current name is very confusing:
```suggestion
        tailoring_file.xpath('//Profile/set-value').each_with_object({}) do |value, obj|
```